### PR TITLE
Add DigitalOcean service discovery

### PR DESCRIPTION
**Pull request by andrewsomething**
This pull request adds DigitalOcean service discovery. Is DigitalOcean "popular enough to justify" inclusion? Well, as the README says, that's an open question. Happy to make the case if you need convincing.

I'd love to see this get included and would appreciate any feedback on its implementation.

Thanks! :christmas_tree:

### CODE REVIEW

File: docs/configuration/configuration.md [hunk 1]
```diff
@@ -330,6 +330,41 @@ Consul setups, the relevant address is in `__meta_consul_service_address`.
...
+* `__meta_digitalocean_instance_size`: A DigitalOcean size slug representing the Droplet's instance type
+* `__meta_digitalocean_region`: A DigitalOcean region slug representing the region where the Droplet is located
+* `__meta_digitalocean_public_ipv4`: The Droplet's public IPv4 address
+* `__meta_digitalocean_private_ipv4`: The Droplet's private IPv4 address
+* `__meta_digitalocean_instance_tags`: A comma separated list of the tags applied to the Droplet
+
+See below for the configuration options for DigitalOcean discovery:
+
+```yaml
+# The information to access the Azure API.
```
Comment (jamtur01) [inline docs/configuration/configuration.md]: Copy-pasta extra?

Comment (andrewsomething) [reply docs/configuration/configuration.md]: :flushed:

**Comment by andrewsomething** [comment]
Rebased on master to resolve the conflicts caused by the changes in #3629 

Also includes a small change to the GCE SD to not pass a nil Context as per staticcheck since I'm updating the vendored version of the context package.

**Comment by metalmatze** [comment]
Just saw this, awesome!
I think DO is big enough to be included. I'd certainly use it, in case this gets merged.

**Comment by krasi-georgiev** [comment]
due to lack of man power to review and maintain new SD we are on permanent hold for adding new SD.

As a replacement we have now added an example how to implement a custom SD adapter that converts any SD to json file that can be used by the file SD 
https://github.com/prometheus/prometheus/tree/master/documentation/examples/custom-sd

**Comment by andrewsomething** [comment]
Looks like a different implementation was finally accepted: https://github.com/prometheus/prometheus/pull/7407

**Comment by roidelapluie** [comment]
I did not even notice this pull request, sorry :-o this pull request was not linked from https://github.com/prometheus/prometheus/issues/4166 (which was created after this one) and I think that is why I missed it.

**Comment by andrewsomething** [comment]
No worries @roidelapluie. Just glad to find out that this is finally supported!

**Pull request closed without merging**